src/fides/api/ops/models/datasetconfig.py

```diff
@@ -69,11 +69,10 @@ def upsert_ctl_dataset(ctl_dataset_obj: Optional[CtlDataset]) -> CtlDataset:
             """
             ctl_dataset_data = data.copy()
             validated_data = Dataset(**ctl_dataset_data.get("dataset", {}))
-
             if ctl_dataset_obj:
                 # It's possible this updates the ctl_dataset.fides_key and this causes a conflict
                 # with another ctl_dataset, if we fetched the datasetconfig.ctl_dataset.
-                for key, val in ctl_dataset_data.get("dataset", {}).items():
+                for key, val in validated_data.dict().items():
```

Comment:
Ah geez sorry about this.  A big theme of the Unified Fides Resources feature https://github.com/ethyca/fides/pull/2254 was this very thing, where we had traditionally been saving CTL Datasets without the extra fields initialized in validation, and I'm introducing another source here 🤦 

good troubleshooting!

Reply:
no need to apologize, this is very deep in the weeds and i'm sure that refactor was a very complex effort. this was a fun one to track down! 😄